src/rust/engine/process_execution/src/local.rs

```diff
@@ -476,21 +498,77 @@ mod tests {
     )
   }
 
+  #[test]
+  fn test_directory_preservation() {
+    let preserved_work_tmpdir = TempDir::new().unwrap();
+    let preserved_work_root = preserved_work_tmpdir.path().to_owned();
+
+    let result = run_command_locally_in_dir(
+      ExecuteProcessRequest {
+        argv: vec![
+          find_bash(),
+          "-c".to_owned(),
+          format!("echo -n {} > {}", TestData::roland().string(), "roland"),
+        ],
+        env: BTreeMap::new(),
+        input_files: fs::EMPTY_DIGEST,
+        output_files: vec![PathBuf::from("roland")].into_iter().collect(),
+        output_directories: BTreeSet::new(),
+        timeout: Duration::from_millis(1000),
+        description: "bash".to_string(),
+      },
+      preserved_work_root.clone(),
+      false,
+    );
+    result.unwrap();
+
+    assert_eq!(preserved_work_root.exists(), true);
+
+    // Collect all of the top level sub-dirs under our test workdir.
+    let subdirs: Vec<PathBuf> = preserved_work_root
+      .read_dir()
+      .expect("failed to read test root")
+      .into_iter()
+      .map(|dirent| dirent.unwrap().path().to_owned())
+      .collect();
+
+    assert_eq!(subdirs.len(), 1);
+
+    // Then look for a file like e.g. `/tmp/abc1234/process-execution7zt4pH/roland`
+    let mut rolands_path = preserved_work_root.clone();
+    rolands_path.push(subdirs.first().unwrap());
+    rolands_path.push("roland");
+
+    assert_eq!(rolands_path.exists(), true);
+  }
+
   fn run_command_locally(
     req: ExecuteProcessRequest,
   ) -> Result<FallibleExecuteProcessResult, String> {
-    run_command_locally_in_dir(req)
+    let work_dir = TempDir::new().unwrap();
```

Comment:
This function is interesting now... `work_dir`, being a `TempDir`, will itself be `rm -rf`'d when this function returns, which vaguely subverts the cleanup flag.

For how this is used today, this is fine, but may be worth a comment for the future. Or maybe not :)

Reply:
yeah, completely intentional for this variant of the helper based on how it's being used.